subsys/net/ip/ipv6_nbr.c

```diff
@@ -866,6 +866,8 @@ enum net_verdict net_ipv6_prepare_for_send(struct net_pkt *pkt)
 	if (net_if_ipv6_addr_onlink(&iface, (struct in6_addr *)ip_hdr->dst)) {
```

Comment:
I have difficulty understanding the commit description, also as this change has nothing to do with ICMPv6, we do not really need to emphasize the ping here, also RA is not relevant here.

Regardless of the minor issues in commit message, when I tried to ping a peer host using its link local address, there was no issue. As the target device address was not in the neighbor cache, zephyr sent solicitation message, and got a neighbor advertisement and then was able to send ICMPv6 Echo-Request to the target.
Why would we need something like this as everything is working ok?

Reply:
> Issue is when we have a default router
> When this default router is backend A and we ping backend B with link local address,
> we will send echo request to backend A instead of B, which shall receive a Redirect and no reply.

I think it makes sense, there's indeed no point checking for routes if we're sending to LL address. It actually surprises me it was not spotted earlier.
This second explaination IMO helps better to understand the reason for this change, could we use it in the commit message instead?

Reply:
Ok, makes sense. I did not had routers configured so did not encounter this problem.

Reply:
updated commit msg